engines/buried/buried.cpp

```diff
@@ -0,0 +1,517 @@
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#include "common/scummsys.h"
+#include "common/config-manager.h"
+#include "common/error.h"
+#include "common/events.h"
+#include "common/fs.h"
+#include "common/savefile.h"
+#include "common/system.h"
+#include "common/textconsole.h"
+#include "common/translation.h"
+#include "engines/util.h"
+#include "gui/message.h"
+
+#include "buried/buried.h"
+#include "buried/console.h"
+#include "buried/database.h"
+#include "buried/frame_window.h"
+#include "buried/graphics.h"
+#include "buried/message.h"
+#include "buried/resources.h"
+#include "buried/sound.h"
+#include "buried/video_window.h"
+#include "buried/window.h"
+
+namespace Buried {
+
+BuriedEngine::BuriedEngine(OSystem *syst, const ADGameDescription *gameDesc) : Engine(syst), _gameDescription(gameDesc) {
+	_gfx = nullptr;
+	_mainEXE = nullptr;
+	_library = nullptr;
+	_sound = nullptr;
+	_timerSeed = 0;
+	_mainWindow = nullptr;
+	_focusedWindow = nullptr;
+	_captureWindow = nullptr;
+	_console = nullptr;
+	_pauseStartTime = 0;
+	_yielding = false;
+
+	const Common::FSNode gameDataDir(ConfMan.get("path"));
+	SearchMan.addSubDirectoryMatching(gameDataDir, "WIN31/MANUAL", 0, 2); // v1.05 era
+	SearchMan.addSubDirectoryMatching(gameDataDir, "WIN95/MANUAL", 0, 2); // v1.10 era (Trilogy release)
+
+	// GOG.com release, plainly extracted files
+	SearchMan.addSubDirectoryMatching(gameDataDir, "data1", 0, 3);
+	SearchMan.addSubDirectoryMatching(gameDataDir, "data2", 0, 3);
+	SearchMan.addSubDirectoryMatching(gameDataDir, "data3", 0, 3);
+}
+
+BuriedEngine::~BuriedEngine() {
+	delete _mainWindow;
+	delete _gfx;
+	delete _mainEXE;
+	delete _library;
+	delete _sound;
+	delete _console;
+
+	// The queue should be empty since all windows destroy their messages
+}
+
+Common::Error BuriedEngine::run() {
+	_console = new BuriedConsole(this);
+
+#ifndef USE_ICONV
+	// The Japanese version needs iconv support
+	if (getLanguage() == Common::JA_JPN)
+		return Common::Error(Common::kUnknownError, "No iconv support available");
+#endif
+
+	if (isTrueColor()) {
+#ifndef USE_RGB_COLOR
+		// Can't play 24bpp version without support
+		return Common::kUnsupportedColorMode;
+#else
+		initGraphics(640, 480, 0);
+
+		if (_system->getScreenFormat().bytesPerPixel == 1)
+			return Common::kUnsupportedColorMode;
+#endif
```

Comment:
It shouldn't be necessary to have an ifdef here, since the screen pixel format check should be sufficient.

```suggestion
		initGraphics(640, 480, 0);

		if (_system->getScreenFormat().bytesPerPixel == 1)
			return Common::kUnsupportedColorMode;
```